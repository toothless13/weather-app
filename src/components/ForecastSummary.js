import iconData from "../data/iconData.json";

const ForecastSummary = (props) => {
  const { date, description, icon, temperature } = props;
  const weatherCode = `${icon.slice(0, 1)}00`;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">{formattedDate}</div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <img src={iconData[weatherCode]} alt="icon showing weather code" />
      </div>
      <div className="forecast-summary__temperature">
        {temperature.max}&deg;C
      </div>
      <div className="forecast-summary__description">{description}</div>
    </div>
  );
};

export default ForecastSummary;
