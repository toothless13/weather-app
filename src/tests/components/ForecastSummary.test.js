// eslint-disable-next-line no-unused-vars
import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary component", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "800",
    temperature: {
      min: 12,
      max: 22,
    },
    onSelect: () => {},
  };

  it("Renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />,
    );

    expect(getByText("Thu Jan 01 1970")).toHaveAttribute(
      "class",
      "forecast-summary__date",
    );
    expect(getByText("Stub description")).toHaveAttribute(
      "class",
      "forecast-summary__description",
    );
    expect(getByTestId("forecast-icon")).toHaveAttribute(
      "class",
      "forecast-summary__icon",
    );
    expect(getByText("22Â°C")).toHaveAttribute(
      "class",
      "forecast-summary__temperature",
    );
  });
});
