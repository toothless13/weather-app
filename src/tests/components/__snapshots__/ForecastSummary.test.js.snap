// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary component Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <img
        alt="icon showing weather code"
        src="http://openweathermap.org/img/wn/01d@2x.png"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
    <button
      type="button"
    >
      More details
    </button>
  </div>
</DocumentFragment>
`;
